docs/apm-mutating-webhook.asciidoc

```diff
@@ -0,0 +1,230 @@
+[[apm-mutating-admission-webhook]]
+== APM mutating admission webhook
+
+:kube-admin-docs: https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/
+:helm-docs:       https://helm.sh/docs/
+
+preview::[]
+
+The APM mutating admission webhook for Kubernetes simplifies the instrumentation and
+configuration of your application pods.
+
+The webhook includes a <<apm-webhook,webhook receiver>> that modifies pods so they are automatically
+instrumented by an Elastic APM agent, and a <<apm-helm-chart,Helm chart>> that manages the webhook
+receiver's lifecycle within Kubernetes.
+
+Learn more below, or skip ahead to <<apm-get-started-webhook>>.
+
+[[apm-webhook]]
+=== Webhook
+
+The webhook receiver modifies pods so they are automatically instrumented by an Elastic APM agent.
+Supported agents include:
+
+// links will be added later
+* Java agent
+* .NET agent
+* Node.js agent
+
+The webhook receiver is invoked on pod creation. After receiving the object definition from the Kubernetes
+API server, it looks through the pod spec for a specific, user-supplied annotation. If found, the pod spec
+is mutated according to the webhook receiver's configuration. This mutated object is then returned to the
+Kubernetes API server which uses it as the source of truth for the object.
+
+[[apm-mutation]]
+==== Mutation
+
+The mutation that occurs is defined below:
+
+. Add an init container image that has the agent binary.
+. Add a shared volume that is mounted into both the init container image and
+all container images contained in the original incoming object.
+. Copy the agent binary from the init container image into the shared volume,
+making it available to the other container images.
+. Update the environment variables in the container images to configure
+auto-instrumentation with the copied agent binary
+
+TIP: To learn more about mutating webhooks,
+see the {kube-admin-docs}[Kubernetes Admission controller documentation].
+
+[[apm-helm-chart]]
+=== Helm chart
+
+The Helm chart manages the configuration of all associated manifest files for the
+webhook receiver, including generating certificates for securing communication
+between the Kubernetes API server and the webhook receiver.
+
+TIP: To learn more about Helm charts, see the {helm-docs}[Helm documentation].
+
+// Break content below to a new page
+
+[[apm-get-started-webhook]]
+== Get started with the webhook
+
+To instrument and configure your application pods, complete the following steps:
+
+. <<apm-webhook-clone-repo>>
+. <<apm-webhook-configure-helm>>
+. <<apm-webhook-install-helm>>
+. <<apm-webhook-add-pod-annotation>>
+// Anything next? Restart the deployment maybe?
```

Comment:
Is there a step we need to add after the pod annotation is added? Like a deployment restart?

Reply:
> Is there a step we need to add after the pod annotation is added? Like a deployment restart?

Seconding this -- once the new configuration with the pod annotation is set that means when an new container/process is started it will be started with the agent auto attachment, but containers/processes already running in pods will not automatically restart AND the addition of the annotation is not the sort of configuration change that will trigger a restart automatically.  Acknowledging this is some way would make sense and a friendly version of "if you want your already running stuff to have agents attached you'll need to restart things".

Reply:
Thanks for the additional context. I think I've added something useful that explains this.